idx/memory/find_cache.go

```diff
@@ -123,50 +142,109 @@ func (c *FindCache) PurgeAll() {
 // disable it for `backoffTime`. Future InvalidateFor calls made during
 // the backoff time will then return immediately.
 func (c *FindCache) InvalidateFor(orgId uint32, path string) {
-	c.RLock()
-	ch := c.newSeries[orgId]
+	c.Lock()
+	defer c.Unlock()
+	if time.Now().Before(c.backoff) {
+		return
+	}
+
 	cache, ok := c.cache[orgId]
-	c.RUnlock()
 	if !ok || cache.Len() < 1 {
 		return
 	}
 
+	req := invalidateRequest{
+		orgId: orgId,
+		path:  path,
+	}
 	select {
-	case ch <- struct{}{}:
+	case c.invalidateReqs <- req:
 	default:
-		c.Lock()
-		c.backoff[orgId] = time.Now().Add(c.backoffTime)
+		log.Infof("memory-idx: findCache invalidate-queue full. Disabling cache for %s. num-cached-entries=%d", c.backoffTime.String(), cache.Len())
+		c.backoff = time.Now().Add(c.backoffTime)
 		delete(c.cache, orgId)
-		c.Unlock()
-		for i := 0; i < len(ch); i++ {
-			select {
-			case <-ch:
-			default:
+		// drain queue
```

Comment:
This is draining all of the invalidateRequests for all orgs, but only the cache for the current Org is deleted.   The caches for the orgs will not get correctly invalidated for the new series in defined in the discarded invalidateRequests.

Reply:
yep. i have changed this to clear caches for all orgs (and backoff state is now also "global", not per org)
too much hassle to try to do everything org by org. not ideal, but good enough for now.